### ISSUE TITLE
Custom icons (PNGs) for Discord Folders

### ISSUE THREAD

**Issue by Ransomwave** [Vencord/plugin-requests#638]
### Plugin Description

Would be nice to be able to **add custom icons to folders on Discord**, so It's easier to tell which folder has what servers (for example, "YouTubers" folder would have a YouTube icon).

There would be a predefined list of icons that go with Discord's design language (eg. using something like fontawesome) and the option to imput a link to a custom image.

They should still be shown as folders if they have custom images without a transparent background by, perhaps, having a colored outline to differenciate it from servers.

Below I will show a mock-up of what I mean. You'll be able to see the regular icon, the predefined icon list, and a custom image:

![foldericons_idea](https://github.com/Vencord/plugin-requests/assets/63156992/746dd2a9-e1ee-4fa3-a42b-4460aac28763)

### Request Agreement

- [X] I read the above and made sure my plugin is possible and allowed
- [X] I checked the [existing plugins](https://vencord.dev/plugins) and made sure my plugin doesn't already exist
- [X] I [searched the existing issues](https://github.com/Vencord/plugin-requests/issues?q=is%3Aissue) and made sure a similar request doesn't already exist

**Comment by Ransomwave** [comment]
There is no other existing Plugin for this.

**Comment by programminglaboratorys** [comment]
isn't this possible with css?

**Comment by Sqaaakoi** [comment]
yeah but it'd be horror to implement probably

**Comment by Ransomwave** [comment]
> isn't this possible with css?

Yes, but you'd have to manually edit every icon every time in the CSS for every single folder you want to change. How about there being a modal that does this for you, in the form of a plugin?

**Comment by Ransomwave** [comment]
> yeah but it'd be horror to implement probably

How so? All it would do is change a few icons in the app, I don't see how this is harder than any of the other plugins that are available in Vencord already.

**Comment by sadan4** [comment]
I made a plugin that does this, but it is very rough around the edges. 
You can find it here <https://github.com/sadan4/Vencord/blob/customFolderIcons/src/plugins/customFolderIcons/index.tsx>
